python/pyspark/sql/functions/builtin.py

```diff
@@ -12227,6 +12258,43 @@ def unhex(col: "ColumnOrName") -> Column:
     return _invoke_function_over_columns("unhex", col)
 
 
+@_try_remote_functions
+def uniform(
+    min: "ColumnOrName", max: "ColumnOrName", seed: Optional["ColumnOrName"] = None
+) -> Column:
+    """Returns a random value with independent and identically distributed (i.i.d.) values with the
+    specified range of numbers. The random seed is optional. The provided numbers specifying the
+    minimum and maximum values of the range must be constant. If both of these numbers are integers,
+    then the result will also be an integer. Otherwise if one or both of these are floating-point
+    numbers, then the result will also be a floating-point number.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    min : :class:`~pyspark.sql.Column`, int, or float
+        Minimum value in the range.
+    max : :class:`~pyspark.sql.Column`, int, or float
+        Maximum value in the range.
+    seed : :class:`~pyspark.sql.Column` or int
+        Optional random number seed to use.
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        The generated random number within the specified range.
+
+    Examples
+    --------
+    >>> spark.createDataFrame([('3',)], ['a']).select(uniform(0, 10).alias('x')).select(isnull('x')).collect()
```

Comment:
```suggestion
    >>> from pyspark.sql import functions as sf
    >>> spark.range(10).select(sf.uniform(sf.lit(0), "id", 123)).show()
```

```
uniform(0, 10)
```
if we want to support int arguments, I think we need to:
1, update the type hint `Union["ColumnOrName", int]`
2, convert them to column by `lit(min)` and `lit(max)`

Reply:
Sounds good, done. Note it was not the intended use case to support column references for any of these arguments; foldable expressions should be sufficient.